test/e2e/specs/test.spec.js

```diff
@@ -1,33 +1,83 @@
-import { mnemonics } from '../test.utils';
+import { mnemonics, phrasesList, MINUTE_MS } from '../test.utils';
+
 describe('Vaults UI Test Cases', () => {
   context('Test commands', () => {
-    it('should set up wallet', () => {
-      // Using exports from the synthetic-chain lib instead of hardcoding mnemonics UNTIL https://github.com/Agoric/agoric-3-proposals/issues/154
-      cy.setupWallet({
-        secretWords: mnemonics.user1,
-        walletName: 'user1',
-      }).then(taskCompleted => {
-        expect(taskCompleted).to.be.true;
-      });
+    const networkPhrases = phrasesList[Cypress.env('AGORIC_NET') || 'local'];
+
+    it('should setup the wallet', () => {
+      if (networkPhrases.isLocal) {
+        cy.setupWallet({
+          secretWords: mnemonics.user1,
+          walletName: 'user1',
+        });
+      } else {
+        const walletAddress = {
+          value: null,
+        };
+        cy.setupWallet({
+          createNewWallet: true,
+          walletName: 'my created wallet',
+        });
+        cy.origin('https://wallet.agoric.app/', () => {
+          cy.visit('/wallet/');
+
+          cy.get('input[type="checkbox"]').click();
+          cy.contains('Proceed').click();
+        });
+        cy.acceptAccess();
+
+        cy.origin('https://wallet.agoric.app/', () => {
+          cy.visit('/wallet/');
+
+          cy.contains(/agoric.{39}/).spread(element => {
+            return element.innerHTML.match(/agoric.{39}/)[0];
+          });
+        }).then(address => {
+          walletAddress.value = address;
+        });
+        cy.origin(
+          'https://emerynet.faucet.agoric.net',
+          { args: { walletAddress } },
+          ({ walletAddress }) => {
+            cy.visit('/');
+            cy.get('[id="address"]').first().type(walletAddress.value);
+            cy.get('[type="submit"]').first().click();
+            cy.get('body').contains('success').should('exist');
+
+            cy.visit('/');
+            cy.get('[id="address"]').first().type(walletAddress.value);
+            cy.get('[type="radio"][value="client"]').click();
+            cy.get('[type="submit"]').first().click();
+            cy.get('body').contains('success').should('exist');
+          },
+        );
+      }
     });
 
+    // eslint-disable-next-line ui-testing/missing-assertion-in-test
     it('should connect with the wallet', () => {
       cy.visit('/');
 
+      if (!networkPhrases.isLocal) cy.contains('button', 'Dismiss').click();
+      cy.get('button').contains('Local Network').click();
+      cy.get('button').contains(networkPhrases.interNetwork).click();
+      if (!networkPhrases.isLocal)
+        cy.get('button').contains('Keep using Old Version').click();
+
       cy.contains('Connect Wallet').click();
```

Comment:
Is 61 to 66 for connecting with emerynet? For local network connection, line 62 and 63 seem redundant? 
Do you think we should do it like:
```js
 if (!networkPhrases.isLocal) {
 cy.contains('button', 'Dismiss').click();
 cy.get('button').contains('Local Network').click();
 cy.get('button').contains(networkPhrases.interNetwork).click(); 
cy.get('button').contains('Keep using Old Version').click();
 }
```

I think by doing this it becomes distinguishable about what's different in local and emerynet

Reply:
`if (!networkPhrases.isLocal) cy.contains('button', 'Dismiss').click();` Do we need this? Haven't encountered this

Reply:
it only occurs when the dapp cant find a local chain, so we dismiss the "Error connecting" dialog box in emerynet tests.
You might not have encountered this if your local chain was running